app/src/main/java/com/kickstarter/ui/activities/ThreadActivity.kt

```diff
@@ -26,28 +26,40 @@ class ThreadActivity :
     private lateinit var ksString: KSString
 
     private val adapter = RepliesAdapter(this)
+    private val linearLayoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, true)
+    private lateinit var recyclerViewPaginator: RecyclerViewPaginator
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         binding = ActivityThreadLayoutBinding.inflate(layoutInflater)
         setContentView(binding.root)
         ksString = environment().ksString()
-
+        recyclerViewPaginator = RecyclerViewPaginator(binding.commentRepliesRecyclerView, { viewModel.inputs.nextPage() }, viewModel.outputs.isFetchingReplies(), false)
+        linearLayoutManager.stackFromEnd = true
         binding.commentRepliesRecyclerView.adapter = adapter
+        binding.commentRepliesRecyclerView.layoutManager = linearLayoutManager
 
         this.viewModel.getRootComment()
```

Comment:
this should be an output: 
instead of `this.viewmodel.getRootComment()` ->`this.viewmodel.output.getRootComment()` same for `this.viewModel.onCommentReplies()`

Reply:
done